src/app.js

```diff
@@ -363,20 +352,43 @@ class Messenger extends EventEmitter {
     // The 'payload' param is a developer-defined field which is set in a postback
     // button for Structured Messages.
     const payload = event.postback.payload;
-
     debug("onPostback for user:%d with payload '%s'", senderId, payload);
-
-    this.emit('text', new Response(this, {event, senderId, session, source: 'postback', text: payload}));
     this.emit('postback', new Response(this, {event, senderId, session, payload}));
+
+    if (this.emitOptionalEvents(event, senderId, session, payload)) {
+      return;
+    }
+    this.emit('text', new Response(this, {event, senderId, session, source: 'postback', text: payload, normalizedText: this.normalizeString(payload)}));
   }
 
   // HELPERS
   //////////
 
+  emitOptionalEvents(event, senderId, session, text) {
+    if (this.options.emitGreetings && this.greetings.test(text)) {
+      const firstName = session.profile && session.profile.first_name.trim() || '';
+      const surName = session.profile && session.profile.last_name.trim() || '';
+      const fullName = `${firstName} ${surName}`;
+
+      this.emit('text.greeting', new Response(this, {event, senderId, session, firstName, surName, fullName}));
+      return true;
+    }
+
+    if (this.help.test(text)) {
+      this.emit('text.help', new Response(this, {event, senderId, session}));
+      return true;
+    }
+    return false;
+  }
+
   getCacheKey(senderId/*: number */)/*: string */ {
     return '' + senderId;
   }
 
+  normalizeString(inputStr) {
```

Comment:
This looks like a good candidate for a function rather than a method. It doesn't rely on any state stored in `this`.